astroid/rebuilder.py

```diff
@@ -119,10 +123,21 @@ def _get_doc(self, node: T_Doc) -> Tuple[T_Doc, Optional[str]]:
                 ):
                     doc = first_value.value if PY38_PLUS else first_value.s
                     node.body = node.body[1:]
-                    return node, doc
+                    if PY38_PLUS:
+                        doc_node = nodes.Const(
+                            value=doc,
+                            lineno=first_value.lineno,
+                            col_offset=first_value.col_offset,
+                            parent=node,
+                            end_lineno=first_value.end_lineno,
+                            end_col_offset=first_value.end_col_offset,
+                        )
```

Comment:
I believe there is an easier way to create a `Const` node. Try
```py
doc_node = self.visit(node.body[0], node)
```
It obviously needs to happen, before the node is removed from the body, but then it should work. Even for Python 3.7.

Furthermore, you should be able to simplify the `doc` definition by using the newly create node.
```py
doc = doc_node.value
```